src/FMU_Wrapper.f90

```diff
@@ -0,0 +1,753 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief Module provides the ability to import and interact with
+!> Functional Mock Up Units (FMU).
+!>
+!> Currently supported FMU versions:
+!>  - 2.0
+!>
+!> The FMU is a zip file contaning an XML file and a *.so library.
+!> The file layout of an unzipped FMU is as follows:
+!>
+!>  FMU_unzipDirectory
+!>   |-> binaries
+!>   |   |-> linux64
+!>   |       |-> modelIdentifier.so
+!>   |-> modelDescription.xml
+!>
+!> Where modelIdentifier and unzipDirectory are specified in the input pList
+!>
+!> See: https://fmi-standard.org for additional info.
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE FMU_Wrapper
+#include "Futility_DBC.h"
+USE Futility_DBC
+USE ISO_C_BINDING
+USE IntrType
+USE ExceptionHandler
+USE ParameterLists
+USE Strings
+USE FMU_interface
+
+IMPLICIT NONE
+PRIVATE
+
+!
+! List of public members
+PUBLIC :: FMU2_Slave
+PUBLIC :: eFMU_Wrapper
+
+!> @brief the base fmu type.
+!> All FMU implementations must provide init, clear, get/set vars and doStep.
+TYPE,ABSTRACT :: FMU_Base
+  !> Initialization status
+  LOGICAL(SBK) :: isInit=.FALSE.
+  !> FMU version
+  INTEGER(SIK) :: FMU_version
+  !> Model id
+  INTEGER(SIK) :: idFMU
+  !> Model globaly unique id
+  TYPE(StringType) :: guid
+  !> Model file name used in FMU XML file open
+  TYPE(StringType) :: modelIdentifier
+  !> Path to unziped FMU
+  TYPE(StringType) :: unzipDirectory
+  !> User defined FMU instance label
+  TYPE(StringType) :: instanceName
+  !> XML derived model description
+  TYPE(ParamType) :: modelDescription
+  !> FMU C opaque pointer to FMU obj
+  TYPE(C_PTR) :: fmu_c_ptr=c_null_ptr
+!
+!List of Type Bound Procedures
+  CONTAINS
+    !> Defered init routine
+    PROCEDURE(fmu_init_sub_absintfc),DEFERRED,PASS :: init
+    !> Defered clear routine
+    PROCEDURE(fmu_clear_sub_absintfc),DEFERRED,PASS :: clear
+    !> Defered setupExperiment routine
+    PROCEDURE(fmu_setupExperiment_sub_absintfc),DEFERRED,PASS :: setupExperiment
+    !> Defered getReal routine
+    PROCEDURE(fmu_getReal_sub_absintfc),DEFERRED,PASS :: getReal
```

Comment:
This capability (to automatically resolve get and set to getReal/Bool/Int or setReal/Bool/Int)  is provided by the `getNamedVariable` and `setNamedVariable` methods.  The getReal, setReal, ect. interfaces are retained as these closely adhear to the FMI standard and may be usuful to a user if only variable handles (valueRefrence) are tracked in the program rather than variable names.